src/wui/http/http_client.c

```diff
@@ -521,24 +529,19 @@ err_t data_received_fun(void *arg, struct tcp_pcb *tpcb, struct pbuf *p, err_t e
         return ERR_ARG;
     }
 
-    if (header_info.content_type == TYPE_INVALID){
-        cmd_status = CMD_REJT_CONT_TYPE;
+    if (false == header_info.valid_request) {
         pbuf_free(p);
+        return ERR_OK;
     }
 
-    if (cmd_status == CMD_UNKNOWN && (HTTPC_RESPONSE_BUFF_SZ < p->tot_len || HTTPC_RESPONSE_BUFF_SZ < header_info.content_lenght)) {
+    if (continue_recv_fun && (HTTPC_RESPONSE_BUFF_SZ < p->tot_len)) {
         cmd_status = CMD_REJT_SIZE;
         result = HTTPC_RESULT_OK;
+        continue_recv_fun = false;
         pbuf_free(p);
     }
 
-    if (cmd_status == CMD_UNKNOWN && (p->tot_len < header_info.content_lenght || p->tot_len > header_info.content_lenght)){
```

Comment:
Why? Content-Length value have to be the same as real content data length to parse properly.

Reply:
please make  a test with it. when i tested it, some response were rejected

Reply:
I ran tested it in "New python tests" task, it passed.